lib/list.rb

```diff
@@ -0,0 +1,13 @@
+class List
+  def list_options(menu)
+    puts "\n========== Libary App =========="
+    puts '+--------------------------------------+'
+    puts '|              DISPLAY MENU             |'
+    puts '+--------------------------------------+'
+    menu.each do |choice, action|
+      puts "| #{choice}. #{action} |"
+    end
+
```

Comment:
- [ ] In your `App.rb` file you have a method called `create_person` and it does not follow the single responsibility principle. To split the `create_person` method into multiple responsibilities and improve its design, you can separate the task of getting user input into a separate method. Here's an example of how you can refactor the code:

```ruby
def create_person
  puts 'Do you want to create a student (1) or a teacher (2)? '
  person_type = gets.chomp.to_i

  case person_type
  when 1
    create_student
  when 2
    create_teacher
  else
    puts 'Person not created!'
  end
end

def create_student
  name = get_user_input('Name: ')
  age = get_user_input('Age: ').to_i
  parent_permission = get_user_input('What\'s the parent permission? (y/n) ').downcase == 'y'

  @people << Student.new(name, age, parent_permission)
  puts 'Student created!'
end

def create_teacher
  name = get_user_input('Name: ')
  age = get_user_input('Age: ').to_i
  specialization = get_user_input('Specialization: ')

  @people << Teacher.new(name, age, specialization)
  puts 'Teacher created!'
end

def get_user_input(prompt)
  print prompt
  gets.chomp
end
```

The `get_user_input` method should use across your application and not just the `app.rb` file. So you could keep it in one of your parent classes so that it is used across your app.